api/types.proto

```diff
@@ -299,20 +324,27 @@ message UpdateStatus {
 		UPDATING = 1;
 		PAUSED = 2;
 		COMPLETED = 3;
-		// TODO(aaronl): add ROLLING_BACK, ROLLED_BACK as part of
-		// rollback support.
+		ROLLBACK_STARTED = 4;
+		ROLLBACK_PAUSED = 5; // if a rollback fails
+		ROLLBACK_COMPLETED = 6;
 	}
 
 	// State is the state of this update. It indicates whether the
-	// update is in progress, completed, or is paused.
+	// update is in progress, completed, paused, rolling back, or
+	// finished rolling back.
 	UpdateState state = 1;
 
 	// StartedAt is the time at which the update was started.
 	Timestamp started_at = 2;
 
-	// CompletedAt is the time at which the update completed.
+	// CompletedAt is the time at which the update completed successfully,
+	// paused, or finished rolling back.
 	Timestamp completed_at = 3;
 
+	// RollbackStartedAt is the time at which the update started rolling
+	// back to the previous version.
+	Timestamp rollback_started_at = 5;
```

Comment:
Yeah, kind of wish we had done this.

I think we can't remove the existing fields. We could add a new `repeated` field to supersede them, but we'd still have to populate the existing timestamp fields when applicable.

So we could do this, but we'd end up with a weird set of fields in the protobuf because we'd have `started_at` and `completed_at` for backward compatibility, and also a repeated field that lists generic events. Do you think it's worth it?